src/compensations/utils/salary.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  isSalariesType,
+  Salaries,
+} from "studio/components/salariesInput/utils/parseSalaries";
+import { SalariesByLocation } from "../../../studio/lib/payloads/compensations";
+import { Result, ResultError, ResultOk } from "../../../studio/utils/result";
+
+export function calculateSalary(
+  examinationYear: number,
+  degree: string,
+  salaries: Salaries,
+): number | undefined {
+  const degreeValue = degree === "bachelor" ? 1 : 0;
+  const adjustedYear = examinationYear + degreeValue;
+  return salaries[adjustedYear];
+}
+
+export function calculatePension(salary: number): number {
+  return Math.round(salary * 0.07);
+}
+
+function examinationYearsFromSalaries(salaries: Salaries) {
+  return Object.keys(salaries).map(Number);
+}
+
+export function minSalariesExaminationYear(salaries: Salaries): number {
+  return Math.min(...examinationYearsFromSalaries(salaries));
+}
+
+export function maxSalariesExaminationYear(salaries: Salaries): number {
+  return Math.max(...examinationYearsFromSalaries(salaries));
+}
+
+export function salariesFromLocation(
+  year: number,
+  locationId: string,
+  salariesByLocation: SalariesByLocation[],
+): Result<Salaries, string> {
+  const yearlySalaries = salariesByLocation.find(
+    (s) => s.location._ref === locationId,
```

Comment:
I'm a big fan of spelling it out
```suggestion
    (salary) => salary.location._ref === locationId,
```